### ISSUE TITLE
Word 2010: Spaces not Announced when Spell Current Line is Activated

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#7439]
STR:

1. Ensuring that Word 2010 is the installed version, open a new document

2. Type the following: "This is a test"

3. Use the "Spell Current Line" gesture

Expected: Assuming that the symbol level is set to "Some", NVDA should output "This space is space a space test"

Actual: The entire line is spoken without any spaces whatsoever

**Comment by ehollig** [comment]
@elliott94, is this still an issue? I am not able to replicate using Word 2016.

**Comment by bhavyashah** [comment]
@ehollig No, everyone should probably be able to reproduce this, because this is less of a bug and more of a change of behaviour for consistency of expectations. The expected result is for the spaces to be announced when the appropriate Symbol level is chosen. A change for the sake of consistency such as this could have severe complications in practical situations, introducing potential confusion for existing users as well as newbies. I would like to kindly vote against the implementation of the requested change.

**Comment by elliott94** [comment]
I disagree; this has nothing to do with symbol levels. The announcement of spaces in Notepad, for example, works as expected without issue whereas in Word 2010 this is not the case.

**Comment by bhavyashah** [comment]
@elliott94  I guess I am a little confused. Pressing the read current line (NVDA+L on a laptop) command once reads the current line like it usually would without explicitly speaking spaces - as tested in Firefox, Word and Notepad. However, the issue I am able to successfully reproduce only on in Word 2010 is when NVDA+L is pressed twice and the current line is spelt but spaces are ignored. Do you think we are now on the same page, or are we still talking about two entirely different issues here?

**Comment by elliott94** [comment]
Apologies; this was entirely my fault, I forgot that one key point! :) Issue title and description changed to reflect this. Thank you for pointing this out!

**Comment by lukaszgo1** [comment]
I cannot reproduce it with Word 2010 32-bit with NVDA version alpha-17928,9dfbd2ee
@elliott94 Are you still experiencing this?

**Comment by elliott94** [comment]
This is still reproduceable for me using both Outlook and Word 2010.

**Comment by lukaszgo1** [comment]
Perhaps I am missing something here. I've tried to reproduce it as follows:
1. Press windows+r type winword and pressed enter.
2. Word 2010 opens.
3. I've typed "This is a test" without quotes.
4. I've pressed NVDA+Up arrow twice quickly to spell the line.

result:
This space  is space a space test

**Comment by elliott94** [comment]
Those steps look good to me. :) Can anybody else reproduce this?

**Comment by lukaszgo1** [comment]
It is getting interesting. I've tried with a clean config of NVDA and was able to reproduce it. After comparing my settings with the default one it looks like that disabling Automatic language switching (when supported) causes spaces to be announced again when spelling line. @elliott94 Can you tried disabling this option and report back?

**Comment by elliott94** [comment]
Confirmed - disabling Automatic Language Switching begins to announce spacing for me as expected. Thank you for the tip!

**Comment by Adriani90** [comment]
This is also reproducible in Office 2016 and newer with NVDA 2023.1 when automatic language switching is enabled.

**Comment by CyrilleB79** [comment]
I have tested with and without UIA, with and without auto language switching.
The issue only occurs when auto language switching is on and Word is using object model.

@seanbudd can you label this as "legacy/community-support" given this only impacts object model?
